### PR TITLE
Initial commit for educational app 'Elliptic Curve Plotter'

### PR DESCRIPTION
**Pull request by kebekus**

### CODE REVIEW

File: de.unifreiburg.ellipticcurve.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        "--device=dri"
+    ],
+    "modules": [
+        {
+            "name": "ellipticcurve",
+            "buildsystem": "cmake-ninja",
+            "config-opts": ["-DCMAKE_BUILD_TYPE=RelWithDebInfo"],
+            "sources": [
+                {
+                    "type": "git",
```
Comment (nedrichards) [inline de.unifreiburg.ellipticcurve.json]: indenting isn't consistent here.

**Comment by nedrichards** [comment]
bot, build de.unifreiburg.ellipticcurve

File: de.unifreiburg.ellipticcurve.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "app-id": "de.unifreiburg.ellipticcurve",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.11",
+    "sdk": "org.kde.Sdk",
+    "command": "ellipticcurve",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=host",
```
Comment (nedrichards) [inline de.unifreiburg.ellipticcurve.json]: what do you need host access for?

**Comment by nedrichards** [comment]
Your appdata ID doesn't appear to match (the ID needs to be the app-id): https://gitlab.com/kebekus/ellipticcurveplotter/blob/master/src/de.unifreiburg.ellipticcurve.appdata.xml#L3

**Comment by kebekus** [comment]
Am Donnerstag, den 11.10.2018, 03:36 -0700 schrieb Nick Richards:
> @nedrichards commented on this pull request.
> In de.unifreiburg.ellipticcurve.json:
> > @@ -0,0 +1,29 @@
> +{
> +    "app-id": "de.unifreiburg.ellipticcurve",
> +    "runtime": "org.kde.Platform",
> +    "runtime-version": "5.11",
> +    "sdk": "org.kde.Sdk",
> +    "command": "ellipticcurve",
> +    "finish-args": [
> +        "--share=ipc",
> +        "--socket=x11",
> +        "--socket=wayland",
> +        "--filesystem=host",
> what do you need host access for?
> 

I need host access because the program can save drawings as PNG or SVG
graphic files.

Best,

Stefan Kebekus.

**Comment by kebekus** [comment]
Am Donnerstag, den 11.10.2018, 03:35 -0700 schrieb Nick Richards:
> @nedrichards commented on this pull request.
> In de.unifreiburg.ellipticcurve.json:
> > +    "command": "ellipticcurve",
> +    "finish-args": [
> +        "--share=ipc",
> +        "--socket=x11",
> +        "--socket=wayland",
> +        "--filesystem=host",
> +        "--device=dri"
> +    ],
> +    "modules": [
> +        {
> +            "name": "ellipticcurve",
> +            "buildsystem": "cmake-ninja",
> +            "config-opts": ["-DCMAKE_BUILD_TYPE=RelWithDebInfo"],
> +            "sources": [
> +                {
> +                    "type": "git",
> indenting isn't consistent here.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

I cannot see what's wrong. Indentation was done automatically by emacs.
Can you please help? -- Thanks.

**Comment by kebekus** [comment]
Am Donnerstag, den 11.10.2018, 03:38 -0700 schrieb Nick Richards:
> Your appdata ID doesn't appear to match (the ID needs to be the app-
> id): 
> https://gitlab.com/kebekus/ellipticcurveplotter/blob/master/src/de.unifreiburg.ellipticcurve.appdata.xml#L3
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

Thank you for pointing this out. I fixed that a few seconds ago.

**Comment by TingPing** [comment]
> I need host access because the program can save drawings as PNG or SVG
graphic files.

You don't need host access to save files. The default Qt file chooser when you prompt a user to save a file should work inside the sandbox.

**Comment by kebekus** [comment]
Am Donnerstag, den 11.10.2018, 17:56 -0700 schrieb TingPing:

> You don't need host access to save files. The default Qt file chooser
> when you prompt a user to save a file should work inside the sandbox.

You're right. I adjusted the permissions accordingly and pushed the
commit.

I am unsure: do you want me to open a new "pull request", or should I
simply wait?

Best wishes & thanks for your work,

Stefan Kebekus.

**Comment by nedrichards** [comment]
bot, build de.unifreiburg.ellipticcurve

**Comment by nedrichards** [comment]
If you look at lines 20-23 in the files view here you'll see that the "type" "url" lines have different indenting to the "tags" and "commit" lines. No need to open a new PR - you can just push updated commits here and we'll continue to review, run test builds etc. e.g. that last build has output here: https://flathub.org/builds/#/builders/2/builds/5285

**Comment by kebekus** [comment]
Am Freitag, den 12.10.2018, 15:58 +0000 schrieb Nick Richards:

> If you look at lines 20-23 in the files view here you'll see that the
> "type" "url" lines have different indenting to the "tags" and
> "commit" lines.

Thank you! I changes all tabs to spaces and thereby fixed the problem.
If  I understand correctly, this should close the last of the issues
that you raised.

Best, Stefan.

**Comment by nedrichards** [comment]
bot, build de.unifreiburg.ellipticcurve

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/de.unifreiburg.ellipticcurve

**Pull request closed without merging**